### PR TITLE
Check Message Hub credentials before creating feed

### PR DESCRIPTION
**Pull request by bjustin-ibm**
Use the Message Hub REST API to list existing topics - this verifies the credentials are correct
While we're at it, make sure the topic they want to subscribe to already exists.

### CODE REVIEW

File: action/messageHubFeed.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function main(params) {
     var namespace = encodeURIComponent(triggerComponents[1]);
     var trigger = encodeURIComponent(triggerComponents[2]);
 
-    var feedServiceURL = 'http://169.46.145.194:5000/triggers/' + namespace + '/' + trigger;
+    var feedServiceURL = 'http://' + params.package_endpoint + '/triggers/' + namespace + '/' + trigger;
```
Comment (bjustin-ibm) [inline action/messageHubFeed.js]: It is expected that this parameter will be bound when the action is installed. See #34.

File: action/messageHubFeed.js [hunk 2]
```diff
@@ -55,6 +62,35 @@ function main(params) {
     }
 }
 
+function checkMessageHubCredentials(params) {
+    // listing topics seems to be the simplest way to check auth
+    var topicURL = params.kafka_admin_url + '/admin/topics';
```
Comment (bjustin-ibm) [inline action/messageHubFeed.js]: This is a new parameter that needs to be supplied when creating the trigger. It is part of the Bluemix credentials and will be "magically" present when using `package refresh`.

File: action/messageHubFeed.js [hunk 3]
```diff
@@ -120,6 +156,20 @@ function validateParameters(rawParams) {
         return;
     }
 
+    if(rawParams.kafka_admin_url) {
+        validatedParams.kafka_admin_url = rawParams.kafka_admin_url;
```
Comment (bjustin-ibm) [inline action/messageHubFeed.js]: This is a new parameter that needs to be supplied when creating the trigger. It is part of the Bluemix credentials and will be "magically" present when using `package refresh`.

Comment (dubeejw) [inline action/messageHubFeed.js]: Update line 119 to hand the case where the body is not valid JSON. Besides that it works for me.

Comment (bjustin-ibm) [inline action/messageHubFeed.js]: @dubeejw The problem was that I was using both `json:true` in the request options _and_ calling `JSON.parse()` on the response body. In this way, `JSON.parse()` is trying to parse an already-parsed JSON object - not a string like it is expecting.

This should be fixed now.

**Comment by dubeejw** [comment]
LGTM.

**Pull request merged**